### ISSUE TITLE
Structural types have no completion of members

### ISSUE THREAD

**Issue by rcano** [scalameta/metals#4021]
### Describe the bug

Structural types show no completions, in the following example:

```scala
  import scala.reflect.Selectable.reflectiveSelectable
  val foo: Any {
    val x: Double
  } = ???

  foo.<completion>
  ```

`x` is never offered despite being a direct member of the type (applyDynamic *is* offered though).

Note: I don't believe this is a feature request, but rather a direct bug, since the type declares the member to be available.

### Expected behavior

members declared by structural types should be suggested by autocompletion.

### Editor/Extension

VS Code

### Version of Metals

v0.11.6

### Extra context or search terms

structural types, missing completions

**Comment by tanishiking** [comment]
@rcano Thank you for reporting! Good news, this is already implemented in the upstream https://github.com/lampepfl/dotty/pull/15283 and will be available with the next Scala3 release :)

**Comment by tanishiking** [comment]
Confirmed this is fixed in 3.2.0-RC1 :tada: stay tuned!
![Screen Shot 2022-06-29 at 13 46 44](https://user-images.githubusercontent.com/9353584/176353810-90a454a6-6a27-4862-b298-4fbe3dc2ce81.png)
